packages/meditrak-server/src/routes/importSurveys/ConfigImporter/processCalculatedConfig.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { getVariables } from '@beyondessential/arithmetic';
+import { splitStringOn, splitStringOnComma } from '../../utilities';
+
+const OPERATOR_TRANSLATIONS = {
+  GT: '>',
+  GE: '>=',
+  LT: '<',
+  LE: '<=',
+  EQ: '=',
+};
+
+export const processCalculatedConfig = async (models, config) => {
+  const calculatedConfig = await processCalculatedConfigByType(models, config);
+  return {
+    type: config.type,
+    ...calculatedConfig,
+  };
+};
+
+const processCalculatedConfigByType = async (models, config) => {
+  switch (config.type) {
+    case 'arithmetic':
+      return translateArithmeticConfig(models, config);
+    case 'condition':
+      return translateConditionConfig(models, config);
+    default:
+      throw new Error(`Invalid calculated type: ${config.type}`);
+  }
+};
+
+const translateArithmeticConfig = async (models, config) => {
+  const codes = getVariables(config.formula);
+
+  let translatedConfig = {
+    formula: await translateFormula(models, config.formula, codes),
+  };
+
+  if (config.defaultValues) {
+    translatedConfig = {
+      ...translatedConfig,
+      defaultValues: await translateDefaultValues(models, config.defaultValues),
+    };
+  }
+
+  if (config.valueTranslation) {
+    translatedConfig = {
+      ...translatedConfig,
+      valueTranslation: await translateValueTranslation(models, config.valueTranslation),
+    };
+  }
+
+  if (config.text) {
+    translatedConfig = {
+      ...translatedConfig,
+      text: await translateText(models, config.text, codes),
+    };
+  }
+
+  return translatedConfig;
+};
+
+/**
+ * formula: (SchFF00Section1Total / SchFF00Section1TotalObtainable) * 100
+ * will be converted to:
+ * formula: (5ed7558b61f76a6ba5003114 / 5ec5bf8961f76a18ad00d4f9) * 100
+ * @param {*} models
+ * @param {*} formula
+ * @param {*} codes
+ */
+const translateFormula = async (models, formula, codes) => {
+  let translatedFormula = formula;
+
+  for (let i = 0; i < codes.length; i++) {
```

Comment:
```suggestion
  for (const code of codes) {
```
We now allow this in our eslint rules :-)

Can do this in a few places if you like. Nothing wrong with using an index though!